### ISSUE TITLE
Running with python3.6 on ubuntu bionic

### ISSUE THREAD

**Issue by cbrewitt** [fzi-forschungszentrum-informatik/Lanelet2#93]
I have followed the instructions in the README to build Lanelet2 using catkin for python 3 on ubuntu bionic. All packages succeed, but when I run `python -c "import lanelet2"`, I get the following error:

```Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/s1686074/projects/catkin_ws/devel/lib/python3/dist-packages/lanelet2/__init__.py", line 12, in <module>
    module = import_module(__name__ + "." + module_name)
  File "/home/s1686074/projects/catkin_ws/lanelet2-env/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: /usr/lib/x86_64-linux-gnu/libboost_python-py27.so.1.65.1: undefined symbol: PyClass_Type```

Do you have any suggestions in how to solve this issue, and run lanelet2 in python 3?

**Comment by poggenhans** [comment]
It seems something went wrong there. The import error shows that lanelet2 was built against the boost python libraries for python 2.7 instead of 3.6.

Here is what I normally do:
```bash
pip3 install --user catkin_pkg
catkin clean -y # to ensure there are no old configuration files left
catkin config --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo -DPYTHON_VERSION=3.6
catkin build -v
```
When you build with `-v` you should see the line `Found PythonInterp: /usr/bin/python3.6 (found version "3.6.9")` somewhere in your build output. If 2.7 is found instead, this would indicate that cmake wasn't configured correctly.

**Comment by cbrewitt** [comment]
> It seems something went wrong there. The import error shows that lanelet2 was built against the boost python libraries for python 2.7 instead of 3.6.
> 
> Here is what I normally do:
> 
> ```shell
> pip3 install --user catkin_pkg
> catkin clean -y # to ensure there are no old configuration files left
> catkin config --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo -DPYTHON_VERSION=3.6
> catkin build -v
> ```
> 
> When you build with `-v` you should see the line `Found PythonInterp: /usr/bin/python3.6 (found version "3.6.9")` somewhere in your build output. If 2.7 is found instead, this would indicate that cmake wasn't configured correctly.

Thank you for the reply. I've tried this, and I see the line following line in multiple places, where it looks like python3.6 is being used:
`Found PythonInterp: /home/cillian/phd/catkin_ws/lanelet-venv/bin/python3.6 (found version "3.6.9")`

Despite this, I still get the same error that I have posted previously. Do you have any other ideas of what could be causing this?

**Comment by janstrohbeck** [comment]
I had the same problem. See this PR for a possible fix: https://github.com/KIT-MRT/mrt_cmake_modules/pull/10

**Comment by poggenhans** [comment]
Yes, seems like its the same issue. Can you check if simply updating mrt_cmake_modules fixes your problem?

**Comment by cbrewitt** [comment]
> Yes, seems like its the same issue. Can you check if simply updating mrt_cmake_modules fixes your problem?

I have tried updating mrt_cmake_modules and building again, but I still have the same issue as before, with the same error message.

**Comment by janstrohbeck** [comment]
As I wrote in the PR, the fix only works for CMake >= 3.11. Maybe you have an older CMake version?

**Comment by cbrewitt** [comment]
> As I wrote in the PR, the fix only works for CMake >= 3.11. Maybe you have an older CMake version? I tested it with CMake 3.14.

Previously I was using CMake 3.10. After upgrading to CMake 3.17, the issue is resolved. Thank you for the help.

**Comment by poggenhans** [comment]
Thanks for the hint. With  kit-mrt/mrt_cmake_modules@a0a53f4 it should now work for 3.10 and below as well.

It's really a pain to get this right. Every version combination of python/boost/cmake seems to require a different approach...
